components/ws-manager/pkg/manager/monitor.go

```diff
@@ -1242,7 +1243,11 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		gitStatus   *csapi.GitStatus
 	)
 	t := time.Now()
-	for i := 0; i < wsdaemonMaxAttempts; i++ {
+
+	// since we're within the content backup loop, the maximum retry time should
+	// rely on the content finalization configuration
+	maxRetry := int(m.manager.Config.Timeouts.ContentFinalization / util.Duration(wsdaemonRetryInterval))
```

Comment:
If time is taken for a process other than `time.Sleep`, will the timeout time(`m.manager.Config.Timeouts.ContentFinalization`) be exceeded?

Reply:
Can we use this?
https://gobyexample.com/timeouts

Reply:
The `wsdaemonMaxAttempts` is an integer that was a value of 24 (120 seconds / 5 seconds), which means its maximum retry is 24 times. Here, I'd like to address a case where the ws-daemon was not reachable (Unavailable) for over 2 minutes.

What you mentioned is another implementation time wait way, but I don't want to bring the change in this PR.

Reply:
I think your implementation is a great improvement even now. My opinion was a bit of an edge case. I will mark the resolution.